src/Form/index.tsx

```diff
@@ -672,8 +674,13 @@ function Form({
         .then(([session, steps]) => {
           updateBackNavMap(session.back_nav_map);
           setIntegrations(session.integrations);
-          if (!isObjectEmpty(initialValues))
+          if (!isObjectEmpty(initialValues)) {
+            const servarKeyToTypeMap = getServarTypeMap(steps);
+            Object.entries(initialValues).map(([key, val]) => {
+              initialValues[key] = castVal(servarKeyToTypeMap[key], val);
```

Comment:
you should maybe store it not in the original user-passed object since we probably dont want to generate unintended side effects for the user